SwiftCompilerSources/Sources/Optimizer/ModulePasses/MandatoryPerformanceOptimizations.swift

```diff
@@ -123,10 +123,17 @@ private func shouldInline(apply: FullApplySite, callee: Function, alreadyInlined
   if callee.isTransparent {
     return true
   }
+
+  if apply.parentFunction.hasOwnership && !callee.hasOwnership {
+    // Cannot inline a non-ossa function into an ossa function
+    return false
+  }
```

Comment:
It would be better to make this check in OptUtils.swift: `FullApplySite.canInline`